include/nbl/asset/ICPUBuffer.h

```diff
@@ -6,73 +6,91 @@
 
 #include <type_traits>
 
-#include "nbl/core/alloc/null_allocator.h"
-
 #include "nbl/asset/IBuffer.h"
 #include "nbl/asset/IAsset.h"
 #include "nbl/asset/IPreHashed.h"
 
 namespace nbl::asset
 {
 
+struct SICPUBufferCreationParams
+{
+    size_t size;
+    void* data = nullptr;
+    size_t alignment = _NBL_SIMD_ALIGNMENT;
+    std::pmr::memory_resource* memoryResource = nullptr;
+};

```

Comment:
should be nested public struct inside `ICPUBuffer`

Reply:
also the memoryResource MUST be the refcounted wrapper type `smart_Refctd_ptr` (then you `ICPUBuffer::create` must take the struct as a r-value reference)